aten/src/ATen/cpu/vec256/functional.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+#include "vec256.h"
+
+namespace at { namespace vec256 {
+namespace {
+
+template <int64_t size>
+inline int64_t _leftover(int64_t x, int64_t y) {
+  if (x + size > y)
+    return y - x;
+  return size;
+}
+
+} // namespace
+
+template <typename scalar_t>
+inline scalar_t reduce_all(
+    Vec256<scalar_t> (*vec_fun)(Vec256<scalar_t>&, Vec256<scalar_t>&),
+    scalar_t* data,
+    int64_t size,
+    scalar_t ident) {
+  Vec256<scalar_t> acc_vec(ident);
+  for (int64_t d = 0; d < size; d += Vec256<scalar_t>::size) {
+    Vec256<scalar_t> value(ident);
+    int64_t leftover = _leftover<Vec256<scalar_t>::size>(d, size);
```

Comment:
Can we just have a regular loop with an epilogue and avoid using `_leftover` altogether? Removing unnecessary branches here is much more valuable than actually templatizing `loads` I think